### PR TITLE
Update Databricks workflow example DAG

### PR DESCRIPTION
**Pull request by vatsrahul1001**
Update Databricks workflow example DAG

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by pankajkoti — APPROVED** [review]

File: tests/system/providers/databricks/example_databricks_workflow.py [hunk 1]
```diff
@@ -32,34 +32,53 @@
...
+#                 "zone_id": "us-east-2b",
+#                 "spot_bid_price_percent": 100,
+#                 "ebs_volume_count": 0,
+#             },
+#             "node_type_id": "i3.xlarge",
+#             "spark_env_vars": {"PYSPARK_PYTHON": "/databricks/python3/bin/python3"},
+#             "enable_elastic_disk": False,
+#             "data_security_mode": "LEGACY_SINGLE_USER_STANDARD",
+#             "runtime_engine": "STANDARD",
+#             "num_workers": 8,
```
Comment (eladkal) [inline tests/system/providers/databricks/example_databricks_workflow.py]: please explain this commented section. Who is we ? who is suppose to use this comment block?

Comment (pankajkoti) [reply tests/system/providers/databricks/example_databricks_workflow.py]: The current example DAG change shows job_cluster_spec for Databricks on Azure.

The commented out piece is an alternative for the users who would like to use Databricks on AWS. I modified the comment a bit.

File: tests/system/providers/databricks/example_databricks_workflow.py [hunk 2]
```diff
@@ -32,34 +32,59 @@
...
+#         },
+#     }
+# ]
+
```
Comment (pankajkoti) [inline tests/system/providers/databricks/example_databricks_workflow.py]:
```suggestion

# job_cluster_spec example for Databricks on Azure
```

File: tests/system/providers/databricks/example_databricks_workflow.py (hunk 1 above)
Comment (eladkal) [reply tests/system/providers/databricks/example_databricks_workflow.py]: So this is a user facing comment. How do we expect users to know that?
We have a way of delivering examples to the docs. It should be a part of it.

Comment (vatsrahul1001) [reply tests/system/providers/databricks/example_databricks_workflow.py]: @eladkal I have removed the comment block and will open a separate PR for AWS example

**Review by pankajkoti — APPROVED** [review]

**Pull request merged**